### PR TITLE
feat: quiz page

### PR DESCRIPTION
**Pull request by tusharkandpal**
To Check:

- added `quiz-context.tsx` & its types in `quiz-context.types.ts`
- created `Rules.tsx` & `Quiz.tsx` page & added styles

**Note:** for use, try out the F.R.I.E.N.D.S quiz for now.

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import "./App.css";
-import { Home } from "./pages/pages";
+import { Home, Rules } from "./pages/pages";
```
Comment (t007rushi) [inline src/App.tsx:2]: pages => index as it won't confuse you just suggestion

File: src/pages/rules/Rules.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+export const Rules = (): JSX.Element => {
+  const { categoryId } = useParams();
+  const {
+    quizState: { activeQuestion, points },
+    quizDispatch,
+  } = useQuiz();
+
+  useEffect(() => {
+    (async () => {
+      const res = await getQuiz(categoryId || "");
```
Comment (t007rushi) [inline src/pages/rules/Rules.tsx:17]: destructure { quizName, questions, points }

**Review by t007rushi — APPROVED** [review]
Overall Looks great 
go ahead merge after suggestions you want to work on.

File: src/pages/rules/Rules.tsx (hunk 2 above)
Comment (tusharkandpal) [reply src/pages/rules/Rules.tsx:17]: it's type is causing issue while destructuring

**Pull request merged**